1-Draft/RFC00XX-PredIctive-IntelliSense.md

```diff
@@ -0,0 +1,359 @@
+creates a negative impact to customers as it would remove all features of PSReadLine. For this
+reason, Predictive IntelliSense has its own PSReadLine configuration to enable and disable
+predictions.
+
+### Enabling and Disabling Predictions
+
+By default, the feature Predictive IntelliSense is disabled. The impact of enabling this feature by
+default may cause new and existing users to become confused at the shell prompt. The feature is
+enabled and disabled with a PSReadLine command Set-PSReadLineOption that is executed at the shell
+prompt or in the user’s Profile.
+
+Predictive IntelliSense currently supports four arguments for a prediction source;
+
+* None - This option disables Predictive IntelliSense
+* History - This option uses only the PSReadLine history for predictions
+* Plugin – This option uses only registered plugin modules for predictions
+* HistoryAndPlugin – This option uses both for predictions
+
+To enable Predictive IntelliSense, enter the following command in the shell or in the users Profile:
+
+```powershell
+Set-PSReadLineOption -PredictionSource History
+```
+
+To disable Predictive IntelliSense, enter the following command in the shell or in the users Profile:
+
+```powershell
+Set-PSReadLineOption -PredictionSource None
+```
+
+### Changing the Prediction View
+
+Predictions are displayed in one of two view depending on the user preference. The default view is InlineView.
+
+* InlineView – This is the default view and displays the prediction inline with the user’s typing.
+  This view is similar to other shells Fish and ZSH.
+* ListView – ListView provides a dropdown list of predictions below the line the user is typing.
+  Users may quickly scan the list, highlight and select the desired prediction.
+
+Users may change the view at the command line, or in the user’s PowerShell profile.
+
+```powershell
+Set-PSReadLineOption -PredictionViewStyle InlineView
+```
+
+### Change the Prediction Color for Accessibility
+
+By default, Predictions appear on the same line the user is typing in a different customizable
+color. To support Accessibility needs, the prediction color is settable in the shell or users
+Profile.
+
+![pi-color](./media/pi-color.png)
+![pi-color](./media/pi-color2.png)
+
+```powershell
+Set-PSReadLineOption -Colors @{ Prediction = '#8A0303'}
+Set-PSReadLineOption -Colors @{ Prediction = '#2F7004'}
+```
+
+Multiple types of color code values are supported in PSReadLine. For more information see
+Set-PSReadLineOption in
+[PSReadLine](https://docs.microsoft.com/en-us/powershell/module/PSReadline/Set-PSReadlineOption?view=powershell-7)
+
+Examples of different color code values:
+
+```powershell
+ # Use a ConsoleColor enum
+Set-PSReadLineOption -Colors @{ Prediction = 'DarkRed'}
+Set-PSReadLineOption -Colors @{ Prediction = [ConsoleColor]::DarkRed}
+
+ # 24 bit color escape sequence
+Set-PSReadLineOption -Colors @{ Prediction = "$([char]0x1b)[38;5;100m"}
```

Comment:
Since prediction requires 7.1 for the plugin model, you can use "`e" to specifying ESC char

Reply:
Leaving as [char]0x1b) for downlevel support at this time.